src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java

```diff
@@ -141,7 +174,77 @@ URL createClickableScanURL(final String scanId) throws MalformedURLException {
             projectId,
             "scans",
             scanId);
-    return new URL(url.getProtocol(), url.getHost(), url.getPort(), path);
+    try {
+      final URL url = new URL(getURL());
+      return new URL(url.getProtocol(), url.getHost(), url.getPort(), path);
+    } catch (final MalformedURLException e) {
+      throw new MojoExecutionException(
+          "Error building clickable Scan URL. Please contact support@contrastsecurity.com for help",
+          e);
+    }
+  }
+
+  /**
+   * Waits for the scan to complete, writes results in SARIF to the file system, and optionally
+   * displays a summary of the results in the console. Translates all errors that could occur while
+   * retrieving results to one of {@code MojoExecutionException} or {@code MojoFailureException}.
+   *
+   * @param operation the scan operation to wait for and retrieve the results of
+   * @throws MojoExecutionException when fails to retrieve scan results for unexpected reasons
+   * @throws MojoFailureException when the wait for scan results operation times out
```

Comment:
I don't think a custom exception type does much for the friendliness of the log message. The message is going to come from the exception's `getMessage()` accessor, right? We either craft a message and pass it to the `MojoFailureException` as we're doing here, or we can make a custom type and craft the message in that type.

Because there's no code that will catch the `ScanTimeoutException` and handle it any differently from `MojoFailureException`, I don't think it makes sense to introduce the new type. However, we could make this message for friendly. Do you think we should include more details in it? We previously print a URL that users can follow to see the Scan in TeamServer. I feel like that should be sufficient for users to follow-up and find out more about the scan if it times out.

Reply:
Logging the url sounds good.